controllers/operands/kubevirt.go

```diff
@@ -458,21 +461,34 @@ func getKVConfig(hc *hcov1beta1.HyperConverged) (*kubevirtcorev1.KubeVirtConfigu
 		}
 	}
 
-	if val, ok := os.LookupEnv(machineTypeEnvName); ok {
-		if val = strings.TrimSpace(val); val != "" {
-			config.MachineType = val
+	var amd64MachineType string
+	// Remain backwards compatibility with the original env variable for a release before removing
+	if machineTypeEnvValue, ok := os.LookupEnv(machineTypeEnvName); ok {
+		amd64MachineType = machineTypeEnvValue
+	} else if machineTypeEnvValue, ok := os.LookupEnv(amd64MachineTypeEnvName); ok {
+		amd64MachineType = machineTypeEnvValue
+	}
 
-			config.ArchitectureConfiguration = &kubevirtcorev1.ArchConfiguration{
-				Amd64: &kubevirtcorev1.ArchSpecificConfiguration{
-					MachineType:      val,
-					OVMFPath:         DefaultOVMFPath,
-					EmulatedMachines: strings.Split(DefaultAMD64EmulatedMachines, ","),
-				},
-				Arm64: &kubevirtcorev1.ArchSpecificConfiguration{
-					MachineType:      val,
-					OVMFPath:         DefaultOVMFPath,
-					EmulatedMachines: strings.Split(DefaultARM64EmulatedMachines, ","),
-				},
+	if amd64MachineType = strings.TrimSpace(amd64MachineType); amd64MachineType != "" {
+		if config.ArchitectureConfiguration == nil {
+			config.ArchitectureConfiguration = &kubevirtcorev1.ArchConfiguration{}
+		}
+		config.ArchitectureConfiguration.Amd64 = &kubevirtcorev1.ArchSpecificConfiguration{
+			MachineType:      amd64MachineType,
+			OVMFPath:         DefaultAMD64OVMFPath,
+			EmulatedMachines: strings.Split(DefaultAMD64EmulatedMachines, ","),
+		}
+	}
+
+	if armMachineType, ok := os.LookupEnv(arm64MachineTypeEnvName); ok {
+		if armMachineType = strings.TrimSpace(armMachineType); armMachineType != "" {
+			if config.ArchitectureConfiguration == nil {
+				config.ArchitectureConfiguration = &kubevirtcorev1.ArchConfiguration{}
+			}
+			config.ArchitectureConfiguration.Arm64 = &kubevirtcorev1.ArchSpecificConfiguration{
```

Comment:
Should we also set `config.architectureConfiguration.defaultArchitecture`?

Reply:
I don't think that is required no, it defaults to the runtime arch during deployment and we only support homogeneous clusters downstream at the moment AFAIK.

Reply:
@acardace apologies for the random ping but can you keep me honest here ^ I don't think there's any need for `config.architectureConfiguration.defaultArchitecture` to be defined by default at the moment.